utils/warnings.ml

```diff
@@ -1138,6 +1145,16 @@ let message = function
   | Generative_application_expects_unit ->
       "A generative functor\n\
        should be applied to '()'; using '(struct end)' is deprecated."
+  | Degraded_to_partial_match ->
+      let[@manual.ref "ss:warn74"] ref_manual = [ 13; 5; 5 ] in
+      Format.asprintf
+        "This pattern-matching appears\n\
+         to be total, but is compiled as partial. \
```

Comment:
My thinking is that two views are reasonable:
- the half-empty glass: we can consider that any match that the compiler cannot prove total should be considered partial, or
- the half-full glass: we can consider that all matches which the type-checker believes are total are "morally total" (the sort of "this never happens in practice" sentiment that explain why everyone was fine leaving a segfault bug open for almost a decade).

I prefer the half-full vision, and choices we made for the *compiler behavior* correspond to this point of view -- not just the compiler messages and documentation. The wording change you suggest comes from the half-empty point of view. But if we seriously decided to adopt the half-empty point of view, we should not just rephrase the documentation, we should change the behavior of the compiler, by raising the standard [partial-match] warning in presence of splits and mutable state.

Reply:
What about leaning more strongly into the half-full vision:

> This pattern-matching is exhaustive (all cases are covered), but is compiled as partial. ...

Reply:
I like "appears to be total". It expresses the fact that the we see the warning as total, but that it might not be. Is there a specific issue with this formulation?

Reply:
My feeling is that "appears to be" fails to convey the reason why the pattern matching appear to be total.
For instance, the pattern matching in
```
type t = A | B | C | D | E | F
let f = function A -> 0 | B -> 1 | D -> 2 | E -> 3 | F -> 4
```
might "appear to be total" for some meaning of "appears to be" too.

Reply:
> This pattern-matching is compiled as partial, even if it appears to be total.
> This typically occurs due to complex matches on mutable fields.

What do you think?

Reply:
That works for me, thanks !